sdks/python/apache_beam/ml/inference/base_test.py

```diff
@@ -39,7 +41,12 @@ class FakeInferenceRunner(base.InferenceRunner):
   def __init__(self, clock=None):
     self._mock_clock = clock
 
-  def run_inference(self, batch: Any, model: Any) -> Iterable[Any]:
+  def run_inference(
+      self,
+      batch: Any,
+      model: Any,
+      prediction_params: Optional[Dict[str, Any]] = None,
```

Comment:
If I take out `prediction_params`, I get 
```
      result_generator = self._inference_runner.run_inference(
>         examples, self._model, self._prediction_params)
E     TypeError: run_inference() takes 3 positional arguments but 4 were given [while running 'RunInference/ParDo(_RunInferenceDoFn)']

apache_beam/ml/inference/base.py:216: TypeError
```

If we allow this not to be a required param, then maybe we can check if `_prediction_params` is `None` like so? 
```
    if not self._prediction_params:
        result_generator = self._inference_runner.run_inference(
            examples, self._model, self._prediction_params)
    else:
            result_generator = self._inference_runner.run_inference(
            examples, self._model)
```
That feels a bit messy though, and breaks our `run_inference` interface, assuming we want to use this modification:
```
class InferenceRunner():
  def run_inference(
      self,
      batch: List[Any],
      model: Any,
      prediction_params: Optional[Dict[str, Any]] = None) -> Iterable[Any]:
    """Runs inferences on a batch of examples and
    returns an Iterable of Predictions."""
    raise NotImplementedError(type(self))
```

Reply:
Heres some code to demonstrate what I mean.

def pytorch_impl(prediction_params = None):
  print ('got prediction_params ' + str(prediction_params))

def tf_impl():
  # Does not have the prediction_params arg, should not crash when called
  # but should crash if called with prediction_params argument
  print('tf_impl')

mode = 'tf'

def base_do_fn(**kwargs):
  if mode == 'tf':
    tf_impl(**kwargs)
  elif mode == 'pytorch':
    pytorch_impl(**kwargs)


def base_run_inference(**kwargs):
  base_do_fn(**kwargs)

#passes
mode='tf'
base_run_inference()

#passes
mode='pytorch'
base_run_inference(prediction_params={'prediction_param': 'value'})

# fails (expected), because we're passing a parameter that TF implementation should not get
mode='tf'
base_run_inference(prediction_params={'prediction_param': 'value'})

----------------
What I'm saying is this method allows forwarding of any named param whatsoever. With what you're doing we'll always be in a spot of modifying an interface and thus having to modify every implementation of the interface going forward for any single parameter that any single implementation wants.

I think here, and going forward, unless named parameters will apply to all models we should not put it into the interface for all models.

Reply:
Ah I see, thanks for the clarification. I was thinking of something else, hence the misunderstanding. I've added `kwargs` to support generalizability.